src/log_handling/RetailLogHandler.ts

```diff
@@ -0,0 +1,344 @@
+        console.debug("[RetailLogHandler] Handling ENCOUNTER_START line:", line);
+
+        if (!this.activity) {
+            super.handleEncounterStartLine(line);
+        } else {
+            const category = this.activity.getCategory();
+            const isChallengeMode = category === VideoCategory.MythicPlus;
+
+            if (isChallengeMode) {
+                const activeChallengeMode = this.activity as ChallengeModeDungeon;
+                const encounterID = parseInt(line.arg(1), 10)
+                const eventDate = line.date();
+                
+                const segment = new ChallengeModeTimelineSegment(
+                    TimelineSegmentType.BossEncounter,
+                    eventDate,
+                    this.getRelativeTimestampForTimelineSegment(eventDate),
+                    encounterID
+                );
+    
+                activeChallengeMode.addTimelineSegment(segment, eventDate);
+                console.debug(`[RetailLogHandler] Starting new boss encounter: ${dungeonEncounters[encounterID]}`)
+            }
+        }
+    }
+
+    handleEncounterEndLine(line: LogLine) {
+        console.debug("[RetailLogHandler] Handling ENCOUNTER_END line:", line);
+
+        if (this.activity) {
+            const category = this.activity.getCategory(); 
+            const isChallengeMode = category === VideoCategory.MythicPlus;
+
+            if (!isChallengeMode) {
+                console.debug("[RetailLogHandler] Must be raid encounter, calling super method.");
+                super.handleEncounterEndLine(line);
+            } else {
+                console.debug("[RetailLogHandler] Challenge mode boss encounter.");
+                const activeChallengeMode = this.activity as ChallengeModeDungeon;
+                const eventDate = line.date();
+                const result = Boolean(parseInt(line.arg(5), 10));
+                const encounterID = parseInt(line.arg(1), 10);
+                const currentSegment = activeChallengeMode.getCurrentTimelineSegment();
+
+                if (currentSegment) {
+                    currentSegment.result = result;
+                }
+
+                const segment = new ChallengeModeTimelineSegment(
+                    TimelineSegmentType.Trash, 
+                    eventDate, 
+                    this.getRelativeTimestampForTimelineSegment(eventDate)
+                )
+
+                // Add a trash segment as the boss encounter ended
+                activeChallengeMode.addTimelineSegment(segment, eventDate);
+                console.debug(`[RetailLogHandler] Ending boss encounter: ${dungeonEncounters[encounterID]}`);
+            }
+        }
+    }
+
+    handleZoneChange(line: LogLine) {
+        console.info("[RetailLogHandler] Handling ZONE_CHANGE line:", line);
+        const zoneID = parseInt(line.arg(1), 10);
+        const isZoneBG = retailBattlegrounds.hasOwnProperty(zoneID);
+
+        if (this.activity) 
```

Comment:
These two cases could probably be split into two private methods that `handleZonechange()` could call to keep the line count and complexity down.

Reply:
Yeah agree I should do this. Pretty ugly in its current form.